### PR TITLE
Added a conditional dendering for the regionMap

### PR DESCRIPTION
**Pull request by pbelokon**
## Description
Added a conditional rendering for RegionMap based on if the region has geometry data, added a button that calls getGeometry with farce set to true when there is not geometry data. 

## Related Issues
Closes: #51 

## How Was This Tested?
This will be tested by loading regions with no geometry data to see if the button appearance. Currently my db works incorrect so I will test this  one I figure that out.

## Checklist
Before submitting your PR, please review the following:
- [x] Commit messages follow the standard template.
- [ ] All commits are signed.
- [x] Related issues are mentioned in the description above.
- [x] I have followed the project's directory structure.
- [x] Linter checks have been passed.

## Summary by CodeRabbit

- **New Features**
	- Introduced a "Fetch Geometry" button for regions in the map component.

- **Enhancements**
	- Improved region geometry fetching with an added force option.

- **Style**
	- Integrated new button styling from Material UI.

### CODE REVIEW

**Comment by pbelokon** [comment]
@OhmSpectator you mentioned that there was a lint script I have to run, where can I find it both frontend and backend are missing it.

**Comment by OhmSpectator** [comment]
@pbelokon, the script can be run from the frontend or backend directory with this command:
```bash
npm run lint
```
I've also updated the CONTRIBUTING.md file:
frontend - https://github.com/OhmSpectator/track-your-regions/blob/main/frontend/CONTRIBUTING.md
backend - https://github.com/OhmSpectator/track-your-regions/blob/main/backend/CONTRIBUTING.md
general - https://github.com/OhmSpectator/track-your-regions/blob/main/CONTRIBUTING.md

**Comment by OhmSpectator** [comment]
>PS D:\track-your-regions\frontend> npm run lint
npm ERR! Missing script: "lint"

You have to pull the latest version of main. You have the old version locally:
Screenshot 2023-12-13 at 00 41 32

**Comment by pbelokon** [comment]
> > PS D:\track-your-regions\frontend> npm run lint
> > npm ERR! Missing script: "lint"
> 
> You have to pull the latest version of main. You have the old version locally: Screenshot 2023-12-13 at 00 41 32

dam didn't forgot to do that thk

**Comment by pbelokon** [comment]
Rebased it

**Comment by pbelokon** [comment]
@OhmSpectator I think there is a mistake in the [front-end](https://github.com/OhmSpectator/track-your-regions/blob/main/frontend/CONTRIBUTING.md) it says that we should run `` npm run linter`` and ``npm run linter:fix"

since in the project we use the below: 
```js
    "lint": "eslint --ext .js,.jsx src",
    "lint:fix": "eslint --fix --ext .js,.jsx src"
```

**Comment by OhmSpectator** [comment]
Hey @pbelokon, I wanted to let you know that I have merged #86 and updated the main branch accordingly. Could you please rebase this PR? Thanks!

**Comment by pbelokon** [comment]
> Hey @pbelokon, I wanted to let you know that I have merged #86 and updated the main branch accordingly. Could you please rebase this PR? Thanks!

Just did it, thank you for teaching me how lol

**Comment by OhmSpectator** [comment]
@coderabbitai review

**Comment by OhmSpectator** [comment]
@pbelokon, I've prepared a prepopulated DB image. You can try to get it for full-setup testing.
I've described how to get it here: https://github.com/OhmSpectator/track-your-regions/blob/main/deployment/README.md#usage-of-prepopulated-database-container
Then you can start the back and the front as described in README (in short: create `.env.development` and fill with proper values and then run `npm run debug` for the backend and `npm run start` for the frontend).

File: frontend/src/components/RegionMap.jsx [hunk 1]
```diff
@@ -120,9 +121,17 @@ function MapComponent() {
...
+    await fetchSelectedRegionGeometry(true);
+    initializeMap();
+  };
+
   return (
     <div>
       {error && <div style={{ color: 'red' }}>{error}</div>}
+      {!fetchSelectedRegionGeometry() && (
+        <Button onClick={handleForceGeometryFetch}>Fetch Geometry</Button>
+      )}
```
[the remark this answers is not in the document]
Comment (OhmSpectator) [reply frontend/src/components/RegionMap.jsx:134]: @pbelokon address this comment, please.

**Review by OhmSpectator — COMMENTED** [review]
@pbelokon, I would also be happy to see that the commit messages are following the template:
```
front: <Topic>.

<Description>

[Issue: #<GitHub Issue Number>]
```

It put this into doc: https://github.com/OhmSpectator/track-your-regions/blob/main/frontend/CONTRIBUTING.md

**Comment by OhmSpectator** [comment]
@pbelokon I've rebased the branch on top of main. So, be careful when starting working on it again: do not forget to pull the latest version.

**Comment by OhmSpectator** [comment]
@pbelokon, I've just realized that the instructions and the scripts to create/deploy the DB are Unix-specific and unsuitable for Windows... I'll try to fix it.

**Pull request closed without merging**